docs/deployment/setup_gardener.md

```diff
@@ -1,44 +1,412 @@
+
+#### Choices
+
+For this example, the following choices are made:
+- [Garden Linux OS extension](https://github.com/gardener/gardener-extension-os-gardenlinux)
+- [Provider Openstack](https://github.com/gardener/gardener-extension-provider-openstack)
+- [Calico](https://github.com/gardener/gardener-extension-networking-calico)
+- [DNS service](https://github.com/gardener/gardener-extension-shoot-dns-service)
+- [Cert service](https://github.com/gardener/gardener-extension-shoot-cert-service)
+- [OIDC service](https://github.com/gardener/gardener-extension-shoot-oidc-service)
+
+Reference documentation:
+- [Extension Contract](https://github.com/gardener/gardener/blob/master/docs/extensions/resources/extension.md#contract-extension-resource)
+- [GEP 01 Extensibility](https://github.com/gardener/gardener/blob/master/docs/proposals/01-extensibility.md)
+- [Extension Library](https://github.com/gardener/gardener/tree/master/extensions#known-extension-implementations)
+
+#### How to install an extension
+
+In order to make an extension known to a Gardener landscape, two resources have to be applied to the virtual Garden cluster - firstly, a `ControllerDeployment` and secondly, a matching `ControllerRegistration`. With both in place, Gardener-managed `ControllerInstallations` take care of the actual deployment of the extension to the target environment.
+
+Check the [controller registration documentation](https://github.com/gardener/gardener/blob/master/docs/extensions/controllerregistration.md) for more details.
+
+Now, the classical way of installing `ControllerDeployment` and `ControllerRegistration` is to prepare the resources and apply them to the virtual Garden cluster. Please note, the `ControllerDeployment` contains a base64 encoded archive of extension's helm chart.
+
+For example, to deploy the [provider Openstack](https://github.com/gardener/gardener-extension-provider-openstack) extension, the [example file](https://github.com/gardener/gardener-extension-provider-openstack/blob/master/example/controller-registration.yaml) can be used.
+
+To simplify this procedure, extensions can be registered through `Extension` resources in the runtime cluster. The Gardener Operator will take care of deploying the required resources to the virtual Garden cluster.
+In addition, this approach allows the extensions and their admission controllers to be deployed to the runtime cluster as well. There, they can provide additional functionality like creating DNS records or `BackupBuckets`.
+More details can be found in the [Extension Resource documentation](https://github.com/gardener/gardener/blob/master/docs/concepts/operator.md#extension-resource).
+
+![extensions](./content/extensions.png)
+
+### Cloud Profile
+
+In order to host any `Shoot` cluster, at least one `CloudProfile` is required. The resource describes supported Kubernetes and OS versions, as well as infrastructure capabilities like regions and their availability zones.
```

Comment:
Just to make it more clear in the beginning that the `CloudProfile` is the resource type.
```suggestion
In order to host any `Shoot` cluster, at least one `CloudProfile` is required. The `CloudProfile` resource describes supported Kubernetes and OS versions, as well as infrastructure capabilities like regions and their availability zones.
```